engine/src/mblandroidmisc.cpp

```diff
@@ -269,7 +269,66 @@ bool MCParseParameters(MCParameter*& p_parameters, const char *p_format, ...)
 
 bool MCSystemPick(MCStringRef p_options, bool p_use_checkmark, uint32_t p_initial_index, uint32_t& r_chosen_index, MCRectangle p_button_rect)
 {
-	return false;
+	// Split the string on new lines
+	MCAutoArrayRef t_options_arrayref;
+	uindex_t noptions = 0;
+	r_chosen_index = 0;
+	
+	MCStringRef *t_options_array = nil;
+	MCPickList *t_pick_list = nil;
+	
+	bool t_success = true;
+	t_success = MCStringSplit(p_options, MCSTR("\n"), nil, kMCCompareExact, &t_options_arrayref);
```

Comment:
Would be more efficient to use `MCStringSplitByDelimiter` which gives a proper list here indeed the canvas module has some inline proper list helpers to fetch as array of... would be nice to move those to foundation and add a string variant. Possibly more than you want to do for this PR though.